### PR TITLE
Log redact on TiFlash and PD (4.0)

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

TiFlash and PD have picked the feature "redact-log" to the release-4.0 branch(released in 4.0.10), but forgot to add documents to the release-4.0 branch.
Add redact log configurations for TiFlash and PD.

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by JaySon-Huang — COMMENTED** [review]
LGTM.
And we should make this PR align with https://github.com/pingcap/docs/pull/4994 ?

**Review by JaySon-Huang — APPROVED** [review]
LGTM

File: pd-configuration-file.md [hunk 1]
```diff
@@ -88,6 +88,12 @@ Configuration items related to security
 + The path of the PEM file that contains the X509 key
 + Default value: ""
 
+### `redact-info-log` <span class="version-mark">New in v4.0.10</span>
+
++ Enables or disables log redaction in PD.
++ If the configuration value is set to `true`, all user data in the PD log will be redacted.
```
Comment (ran-huang) [inline pd-configuration-file.md:94]: Do you think we should align this with v5.0? https://docs.pingcap.com/tidb/v5.0/pd-configuration-file#redact-info-log-new-in-v500-rc
The description is not consistent.

**Review by ran-huang — APPROVED** [review]

File: pd-configuration-file.md (hunk 1 above)
Comment (TomShawn) [reply pd-configuration-file.md:94]: This is made inconsistent on purpose. This `redact-info-log` is introduced in v5.0.0-rc for dev/v5.0 and in v4.0.10 for v4.0. v5.0.0-rc is earlier than v4.0.10.

Comment (TomShawn) [inline pd-configuration-file.md:94]:
```suggestion
+ Controls whether to enable log redaction in the PD log.
+ When you set the configuration value to `true`, user data is redacted in the PD log.
```

**Pull request merged**